.github/workflows/quarkus-nightly.yml

```diff
@@ -0,0 +1,78 @@
+name: Quarkus Nightly
+
+on:
+  schedule:
+    - cron: '0 0 * * *'
+  push:
+    branches:
+      - quarkus-nightly
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  # Only cancel jobs for PR updates
+  group: quarkus-nightly-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  rebase-quarkus-nightly-branch:
+    name: Rebase quarkus-nightly branch
+    if: github.event_name != 'schedule' || github.repository == 'keycloak/keycloak'
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: quarkus-nightly
+          fetch-depth: 0
+
+      - name: Configure Git
+        run: |
+          git config --global user.name "GitHub Actions"
+          git config --global user.email "github-actions@github.com"
+
+      - name: Fetch and rebase
+        run: |
+          git fetch origin quarkus-nightly
+          git rebase origin/main
+          if [ "$(grep -oP '(?<=<quarkus.version>).*?(?=<\/quarkus.version>)' ${GITHUB_WORKSPACE}/pom.xml)" != "999-SNAPSHOT" ]; then
+            echo "Error: <quarkus.version> in pom.xml is not set to 999-SNAPSHOT."
+            exit 1
+          fi
+
+      - name: Push changes
+        run: |
+          git push -f origin HEAD:quarkus-nightly
+
+  run-nightly-quarkus:
+    name: Run workflows with the nightly Quarkus release
+    runs-on: ubuntu-latest
```

Comment:
Workflows already run automatically on pushes to a branch, so you don't need to trigger this separately here:
https://github.com/keycloak/keycloak/blob/8cd40bd911566ce8a464e24186914b33b7c04676/.github/workflows/ci.yml#L5-L7

Reply:
Initially we discussed that just `ci` and  `operator-ci` are just enough to run with quarkus-next.
As per suggestion I removed the matrix trigger, but at the same time I added the quarkus-next branch among on-push ignored branches in the rest of the workflows. Also, I removed on-push trigger from the quarkus-next workflow itself (to prevent possible infinite loop).

The reason why I used the manual trigger in the previous iteration was that the `git push` is by default executed with `secret.GITHUB_TOKEN`, which, according to the [documentation](https://docs.github.com/en/enterprise-server@2.22/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow), does not trigger a (subsequent) workflow run.

In we want to rely on the on-push triggers, we need to use/create a different token (ideally owned by the organization/created under the keycloak/keycloak repo - I don't have permissions to do it).

cc: @vmuzikar